### PR TITLE
[Form Field v2] Tooltip for FF Helper text

### PR DESCRIPTION
**Pull request by libertymayc**
Tooltip integration with FF for helper text in low screen real estate scenarios
Part of V2 Form Field features

### CODE REVIEW

**Review by alycrys — APPROVED** [review]

File: packages/core/stories/form-field/form-field.doc.stories.mdx [hunk 1]
```diff
@@ -70,6 +70,12 @@ Form Field can display Helper Text using the `FormFieldHelperText` component.
   <Story id="core-form-field--helper-text" />
 </Canvas>
 
+If you are low on screen real-estate, opt for using a Tooltip to display your helper text.
+
+<Canvas>
+  <Story id="core-form-field--helper-text-as-tooltip" />
+</Canvas>
+
```
[the remark this answers is not in the document]
Comment (libertymayc) [reply packages/core/stories/form-field/form-field.doc.stories.mdx:78]: good idea - @bhoppers2008 to advise

[the remark this answers is not in the document]
Comment (libertymayc) [reply packages/core/stories/form-field/form-field.doc.stories.mdx:78]: can add in a separate pr though as will need to add a lot more documentation once ben gets around to the specs for Input v2 too

**Comment by bhoppers2008** [comment]
@libertymayc do we have any control over tooltip position? I notice when width is restricted the tooltip is clipped, rather than appearing above or below.

Also, is there an agreed standard for tooltip reveal? Specifically:

- The delay on keyboard focus - appears instantly (is that common convention with focus?
- Animation on reveal - i thought the tooltip was supposed to have an animation in/out of some description? @navkaur76 @alycrys ?

**Comment by alycrys** [comment]
> 

- Tooltip is clipped - this shouldn't happen. @bhoppers2008 could you please point out the story (steps to follow) where this is happening?
- Delay on keyboard focus - floating-ui has a delay option only for hover. would take some effort to add it for keyboard focus and not sure it adds much value, as keyboard users don't normally care for animations and visual delays.
- Animation on reveal - was meant to come later as an enhancement

**Comment by bhoppers2008** [comment]
> > 
> 
> * Tooltip is clipped - this shouldn't happen. @bhoppers2008 could you please point out the story (steps to follow) where this is happening?
> * Delay on keyboard focus - floating-ui has a delay option only for hover. would take some effort to add it for keyboard focus and not sure it adds much value, as keyboard users don't normally care for animations and visual delays.
> * Animation on reveal - was meant to come later as an enhancement

@alycrys 
Story: https://c8451ff8.saltdesignsystem-storybook.pages.dev/iframe?args=&id=core-form-field--helper-text-as-tooltip&viewMode=story
1. Open dev tools
2. Drag dev tools window so reduced space to either side of form field
3. Hover over FF
4. Tooltip appears clipped to left or right, rather than above (first choice) or below (second choice).
5. Another observation... alternate hover, tooltip moves from left to right. On right site, text wraps. On left side, text is on single line (both clipped).

On KB focus, works for me. Good to have a clear reason 👍

Animation... understood. Is there an issue to track it?

**Comment by alycrys** [comment]
> > > 
> > 
> > 
> > 
> > * Tooltip is clipped - this shouldn't happen. @bhoppers2008 could you please point out the story (steps to follow) where this is happening?
> > * Delay on keyboard focus - floating-ui has a delay option only for hover. would take some effort to add it for keyboard focus and not sure it adds much value, as keyboard users don't normally care for animations and visual delays.
> > * Animation on reveal - was meant to come later as an enhancement
> 
> @alycrys Story: https://c8451ff8.saltdesignsystem-storybook.pages.dev/iframe?args=&id=core-form-field--helper-text-as-tooltip&viewMode=story
> 
> 1. Open dev tools
> 2. Drag dev tools window so reduced space to either side of form field
> 3. Hover over FF
> 4. Tooltip appears clipped to left or right, rather than above (first choice) or below (second choice).
> 5. Another observation... alternate hover, tooltip moves from left to right. On right site, text wraps. On left side, text is on single line (both clipped).
> 
> On KB focus, works for me. Good to have a clear reason 👍
> 
> Animation... understood. Is there an issue to track it?

@bhoppers2008 this is only an issue if the screen is smaller than 365px, but not an issue if there is a smaller wrapper. As we're not dealing with the screen size at this point, I wouldn't count this as an issue for us to fix.
Don't think there is an issue for the animation. Maybe we should have one if that's definitely something we want to do

**Pull request merged**